src/Marketplace.sol

```diff
@@ -304,7 +304,10 @@ contract MarketPlace {
         ERC5095 pt = ERC5095(markets[u][m][uint8(Principals.Illuminate)]);
 
         // Set the pool for the principal token
-        pt.setPool(a);
+        pt.setPool(address(pt));
+
+        // Approve the marketplace to spend the principal and underlying tokens 
+        pt.approveMarketPlace();
```

Comment:
So another thought could be to put this into the creator.sol method and call it there after the initial PT is created.

Though it may be reasonable to leave how youve implemented this here with the heuristic that "setting the pool enables all the rest of the 5095 methods"